app/assets/src/components/visualizations/Histogram.js

```diff
@@ -271,6 +336,18 @@ export default class Histogram {
       }
     }
 
+    this.midpointsToDataIndices = midpointsToDataIndices;
+    this.sortedMidpoints = midpoints.sort((a, b) => a - b);
+
+    this.svg
```

Comment:
Are you doing that refactor soon? I see we perpetuating this appending without data binding issue...